synapse/storage/types.py

```diff
@@ -65,5 +65,5 @@ def rollback(self, *args, **kwargs) -> None:
     def __enter__(self) -> "Connection":
         ...
 
-    def __exit__(self, exc_type, exc_value, traceback) -> bool:
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
```

Comment:
maybe this should be `Optional[bool]` ?

Reply:
i was *just* thinking about that, maybe it should, yeah